caas/kubernetes/provider/k8s.go

```diff
@@ -2619,59 +2645,89 @@ func operatorConfigMap(appName, operatorName string, config *caas.OperatorConfig
 	}
 }
 
-type unitSpec struct {
+type workloadSpec struct {
 	Pod     core.PodSpec `json:"pod"`
-	Service *K8sServiceSpec
+	Service *specs.ServiceSpec
+
+	ServiceAccount            *specs.ServiceAccountSpec
+	CustomResourceDefinitions map[string]apiextensionsv1beta1.CustomResourceDefinitionSpec
 }
 
-func makeUnitSpec(appName, deploymentName string, podSpec *caas.PodSpec) (*unitSpec, error) {
+func processContainers(deploymentName string, podSpec *specs.PodSpec, spec *core.PodSpec) error {
+
+	type containers struct {
+		Containers     []specs.ContainerSpec
+		InitContainers []specs.ContainerSpec
+	}
+
+	var cs containers
+	for _, c := range podSpec.Containers {
+		if c.Init {
+			cs.InitContainers = append(cs.InitContainers, c)
+		} else {
+			cs.Containers = append(cs.Containers, c)
+		}
+	}
+
 	// Fill out the easy bits using a template.
 	var buf bytes.Buffer
-	if err := defaultPodTemplate.Execute(&buf, podSpec); err != nil {
-		return nil, errors.Trace(err)
+	if err := defaultPodTemplate.Execute(&buf, cs); err != nil {
+		logger.Errorf("unable to execute template for containers: %+v, err: %+v", cs, err)
```

Comment:
Unless we really need to log, better to errors.Annotate() the error with this message